src/NuGet.Core/NuGet.Common/PreviewFeatureSettings.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Common
+{
+    // TODO: Probably a better place to put this
+    public static class PreviewFeatureSettings
```

Comment:
I'm not married to this as a name, or a location, or a pattern. I just wanted a common location for doing the actual environment read. Totally open to a suggestion of a better place to put this.